tests/run.go

```diff
@@ -0,0 +1,1209 @@
+// +build ignore
+
+// skip
+
+// Copyright 2012 The Go Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Run runs tests in the test directory.
+//
+// To run manually with summary, verbose output, and full stack traces of of known failures:
+//
+// 	go run run.go -summary -v -show_known_fails
+//
+// TODO(bradfitz): docs of some sort, once we figure out how we're changing
+// headers of files
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"hash/fnv"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+	"unicode"
+)
+
+// -----------------------------------------------------------------------------
+// GOPHERJS: Known test fails for GopherJS compiler.
+//
+// TODO: Reduce these to zero or as close as possible.
+//
+var knownFails = map[string]failReason{
+	// GopherJS compiler panics.
+	"fixedbugs/bug180.go": {category: compilerPanic, desc: `
+panic: 20
+
+goroutine 1 [running]:
+github.com/gopherjs/gopherjs/compiler.(*funcContext).fixNumber(0xc820164210, 0xc82000f360, 0x9088a0, 0xc820018ea0)
+	/Users/Dmitri/Dropbox/Work/2013/GoLand/src/github.com/gopherjs/gopherjs/compiler/expressions.go:1176 +0x235
```

Comment:
Whoops, didn't see this comment. I agree, let's make the descriptions nicer in a followup commit.

Although I find it I end up just trying to investigate and fix the bug if possible right away... E.g., https://github.com/gopherjs/gopherjs/pull/298.